packages/shared/src/components/post/block/PostTagsPanel.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { useBlockPost } from '../../../hooks/post/useBlockPost';
+import { Post } from '../../../graphql/posts';
+import { isNullOrUndefined } from '../../../lib/func';
+import { PostBlockedPanel } from './PostBlockedPanel';
+import CloseButton from '../../CloseButton';
+import { Button, ButtonSize } from '../../buttons/Button';
+import { SourceAvatar } from '../../profile/source';
+import useFeedSettings from '../../../hooks/useFeedSettings';
+import { BlockTagSelection, getBlockedMessage } from './common';
+
+interface PostTagsPanelProps {
+  post: Post;
+  className?: string;
+  toastOnSuccess?: boolean;
+}
+
+export function PostTagsPanel({
+  post,
+  className,
+  toastOnSuccess = true,
+}: PostTagsPanelProps): ReactElement {
+  const { feedSettings } = useFeedSettings();
+  const hasBlockedSource = () =>
+    feedSettings?.excludeSources?.some(({ id }) => id === post.source.id);
+  const [initialPreference] = useState(hasBlockedSource);
```

Comment:
Do you need a useState if there is no setter?
Seems like `hasBlockedSource` itself would do it right?

Reply:
Yes. We need it here so it never changes, because once they have clicked the "Block" button, they would then have the updated value if it would come from the function.

Reply:
Oh I see :) Ok